realm/src/main/java/io/realm/Realm.java

```diff
@@ -1872,4 +1890,24 @@ public static Object getDefaultModule() {
         void execute(Realm realm);
     }
 
+    /**
+     * Encapsulates an async {@link RealmQuery}.
+     * <p>
+     * This will run the {@link RealmQuery} on a worker thread, then invoke this callback on the caller thread
+     */
+    public interface AsyncCallback<T extends RealmResults<? extends RealmObject>> {
```

Comment:
Any reason you are prefixing everything with ```Async```? I don't think I have seen that done in other frameworks as Callback implies asynchronous behaviour anyway ? I would probably call this ```QueryCallback``` or just ```Callback``?